tests/test_unit/test_cdf_tk/test_commands/test_pull.py

```diff
@@ -519,3 +524,71 @@ def test_to_write_content(
         )
         assert not extra_files, "This tests does not support testing extra files"
         assert actual.splitlines() == expected.splitlines()
+
+    def test_pull_transformation_with_sql(
+        self,
+        tmp_path: Path,
+        toolkit_client_approval: ApprovalToolkitClient,
+        env_vars_with_client: EnvironmentVariables,
+    ) -> None:
+        tmp_org_path = tmp_path / "org"
+        tmp_org_path.mkdir()
+        shutil.copytree(
+            str(ORG_FOR_PULL / "modules" / "my_placeholder_module"),
+            str(tmp_org_path / "modules" / "my_placeholder_module"),
+        )
+
+        tmp_build_path = tmp_org_path / "build"
+        tmp_build_path.mkdir()
+
+        toolkit_client_approval.mock_client.transformations.retrieve_multiple.side_effect = (
+            lambda ids, external_ids, ignore_unknown_ids: (
+                TransformationList(
+                    resources=[
+                        Transformation(
+                            external_id="placeholder",
+                            name="pump_asset_hierarchy-load-collections_pump",
+                            destination=TransformationDestination(type="asset_hierarchy"),
+                            isPublic=True,
+                            conflictMode="upsert",
+                            query="SELECT * FROM my_table",
+                        )
+                    ]
+                )
+                if external_ids == ["placeholder"]
+                else TransformationList(resources=[])
+            )
+        )
```

Comment:
We already have this builtin 
```suggestion
        cdf_transformation = Transformation(
                            external_id="placeholder",
                            name="pump_asset_hierarchy-load-collections_pump",
                            destination=TransformationDestination(type="asset_hierarchy"),
                            isPublic=True,
                            conflictMode="upsert",
                            query="SELECT * FROM my_table",
                        )
        toolkit_client_approval.append(Transformation, cdf_transformation)
```